src/convertToObject.js

```diff
@@ -6,7 +6,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const properties = sourceString.split(';');
+  const result = {};
+
+  properties.forEach((property) => {
```

Comment:
Using `forEach` to iterate over properties and build the result object is acceptable and aligns with the checklist exception for not modifying an existing array/object.